admin/release.py

```diff
@@ -303,3 +327,179 @@ def publish_docs_main(args, base_path, top_level):
+    :param bytes source_repo: Repository on the build server to get packages
+        from.
+    :param list packages: List of bytes, each specifying the name of a package
+        to upload to the repository.
+    :param bytes flocker_version: The version of flocker to upload packages
+        for.
+    :param distro_name: The name of the distribution to upload packages for.
+    :param distro_version: The distro_version of the distribution to upload
+        packages for.
+    """
+    rpm_directory.createDirectory()
+
+    yield Effect(DownloadS3KeyRecursively(
+        source_bucket=target_bucket,
+        source_prefix=target_key,
+        target_path=rpm_directory,
+        filter_extensions=('.rpm',)))
+
+    downloaded_packages = yield Effect(DownloadPackagesFromRepository(
+        source_repo=source_repo,
+        target_path=rpm_directory,
+        packages=packages,
+        flocker_version=flocker_version,
+        distro_name=distro_name,
+        distro_version=distro_version,
+        ))
+
+    existing_metadata = yield Effect(
+        ListS3Keys(bucket=target_bucket,
+                   prefix=os.path.join(target_key, 'repodata/')))
+
+    changed_metadata = yield Effect(CreateRepo(
+        repository_path=rpm_directory,
+        existing_metadata=existing_metadata,
+        ))
+
+    yield Effect(UploadToS3Recursively(
+        source_path=rpm_directory,
+        target_bucket=target_bucket,
+        target_key=target_key,
+        files=downloaded_packages | changed_metadata,
+        ))
+
+
+@do
+def upload_rpms(scratch_directory, target_bucket, version, build_server):
+    """
+    Upload RPMS from build server to yum repository.
+
+    :param FilePath scratch_directory: Temporary directory to download
+        repository to.
+    :param bytes target_bucket: S3 bucket to upload repository to.
+    :param bytes version: Version to download RPMs for.
+    :param bytes build_server: Server to download new RPMs from.
+    """
+    if not (is_release(version)
+            or is_weekly_release(version)):
+        raise NotARelease
+
+    if get_doc_version(version) != version:
+        raise DocumentationRelease
+
+    if is_release(version):
+        target_distro_suffix = ""
+    elif is_weekly_release(version):
+        target_distro_suffix = "-testing"
+
+    yield update_repo(
+        rpm_directory=scratch_directory.child(b'fedora-20-x86_64'),
+        target_bucket=target_bucket,
+        target_key=os.path.join(b'fedora' + target_distro_suffix, b'20',
+                                b'x86_64'),
+        source_repo=os.path.join(build_server, b'results/omnibus', version,
+                                 b'fedora-20'),
+        packages=FLOCKER_PACKAGES,
+        flocker_version=version,
+        distro_name='fedora',
+        distro_version='20',
+    )
+
+    yield update_repo(
+        rpm_directory=scratch_directory.child(b'centos-7-x86_64'),
```

Comment:
This seems like we could loop over the list `[('fedora', '20', 'x86_64'), ('centos', '7', 'x86_64')]`.